homeassistant/components/tessie/entity.py

```diff
@@ -74,8 +118,70 @@ async def run(
                 translation_placeholders={"name": name},
             )
 
-    def set(self, *args: Any) -> None:
-        """Set a value in coordinator data."""
-        for key, value in args:
-            self.coordinator.data[key] = value
-        self.async_write_ha_state()
+    def _async_update_attrs(self) -> None:
+        """Update the attributes of the entity."""
+        # Not used in this class yet
+
+
+class TessieEnergyInfoEntity(TessieBaseEntity):
+    """Parent class for Tessie energy site info entities."""
+
+    def __init__(
+        self,
+        data: TessieEnergyData,
+        key: str,
+    ) -> None:
+        """Initialize common aspects of a Tessie energy site info entity."""
+
+        self._attr_unique_id = f"{data.id}-{key}"
+        self._attr_device_info = data.device
+
+        super().__init__(data.info_coordinator, key)
+
+
+class TessieEnergyLiveEntity(TessieBaseEntity):
+    """Parent class for Tessie energy site live entities."""
+
+    def __init__(
+        self,
+        data: TessieEnergyData,
+        key: str,
+    ) -> None:
+        """Initialize common aspects of a Tessie energy site live entity."""
+
+        self._attr_unique_id = f"{data.id}-{key}"
+        self._attr_device_info = data.device
+
+        super().__init__(data.live_coordinator, key)
```

Comment:
I am now thinking, these 2 classes don't really differ, except for the fact that one uses a different coordinator. But because these are 2 different classes, we also need to have 2 sensor entities, and they also look exactly the same. Would it make sense to merge these 2 and make the `data.id` and `coordinator` a parameter solve this? For the device, you can make use of `dr.async_get_or_create` to just create a device with the properties, and then having `self._attr_device_info = DeviceInfo(identifiers={(DOMAIN, data.id)}` would link everything together. For an example check lidarr (got merged today) or aosmith

Reply:
You ask would it make sense, and I'd turn that around and ask you if it makes sense?

As I see it, having two classes means I am sure what type of entity it is, type checking will ensure it has the right coordinator, and the code is easier to understand because there is a clear distinction between the two similar but different data sources.

Is there benefit in reducing 14 lines of code in the entity file to make the other platform init a touch more complex.

The only other aspect from a maintainance point of this, is if I make this change here, I probably need to make it in Teslemetry too otherwise the code patterns will be unique in subtle ways that will make it harder to maintain both (since they are functionally equivilant).

I can make the change if you think its the best outcome.

Reply:
I have just pushed a change implementing this.